stake-pool/program/src/processor.rs

```diff
@@ -1958,9 +1975,11 @@ impl Processor {
             .pool_token_supply
             .checked_add(new_pool_tokens)
             .ok_or(StakePoolError::CalculationFailure)?;
+        // We treat the extra lamports as though they were
+        // transferred directly to the reserve stake account.
         stake_pool.total_stake_lamports = stake_pool
             .total_stake_lamports
-            .checked_add(all_deposit_lamports)
+            .checked_add(credited_deposit_lamports)
```

Comment:
Since we're withdrawing `additional_lamports` into the reserve, and the reserve lamports are part of `total_stake_lamports`, this should be `all_deposit_lamports` still

Reply:
technically yes, but I wanted it to be lazily updated by `update_pool_balance` since I wanted it to be equivalent to directly transferring lamports into the reserve account - no one knows about that until the next update.

Reply:
If someone deposits too much into the stake pool, it gets more valuable right away, and not at the start of the next epoch, right?  If someone withdraws right after this, they should get the right amount, so I say go with `all_deposit_lamports`

Reply:
Ok, fair enough